doc/api/cluster.markdown

```diff
@@ -88,30 +116,338 @@ Node.js process and a cluster worker differs:
    want to listen on a unique port, generate a port number based on the
    cluster worker ID.
 
-There is no routing logic in Node.js, or in your program, and no shared
-state between the workers.  Therefore, it is important to design your
-program such that it does not rely too heavily on in-memory data objects
-for things like sessions and login.
+It is important to keep in mind that worker processes do not share state or
+memory with each other, or with the master, beyond the shared socket handles
+used for distributing connection workload. Applications using cluster must be
+written in such a way that they do not rely too heavily on in-memory data
+objects for things like sessions and login.
+
+Because workers are all separate processes, it is safe to terminate and
+re-spawn individual workers depending on the program's needs, without
+affecting any of the other workers. As long as there are some workers still
+alive, the server will continue to accept connections.  If no workers are
+alive, existing connections will be dropped and new connections will be
+refused.  Node.js does not automatically manage the number of workers for you,
+however.  It is your responsibility to manage the worker pool for your
+application's needs.
+
+## The `cluster` module
+
+The `cluster` module is loaded using `require('custer')`:
+
+    const cluster = require('cluster');
+
+The `cluster` module itself is an instance of `'EventEmitter'` and exposes a
+number of properties and methods whose values and functions will vary depending
+on if the current process is the master or a worker.
+
+### Event: 'disconnect'
+
+* `worker` {Worker object}
+
+Emitted within the master process after a worker IPC channel has disconnected.
+This can occur when a worker exits gracefully, is killed, or is disconnected
+manually (such as when using `worker.disconnect()`).
+
+There can be a delay between the `'disconnect'` and `'exit'` events.  These
+events can be used, for instance, to detect when a working process is stuck in
+a cleanup process or if there are long-living connections.
+
+When triggered, the callback function will be passed the `Worker` instance
+representing the worker that has disconnected:
+
+    cluster.on('disconnect', (worker) => {
+      console.log(`The worker #${worker.id} has disconnected`);
+    });
+
+### Event: 'exit'
+
+* `worker` {Worker object}
+* `code` {Number} the exit code, if it exited normally.
```

Comment:
"terminated normally" or just "exited" would be more consistent with child process docs